src/EFCore.PG/Query/Internal/NpgsqlSqlNullabilityProcessor.cs

```diff
@@ -276,6 +278,38 @@ PostgresUnknownBinaryExpression postgresUnknownBinaryExpression
         return jsonTraversalExpression.Update(expression, newPath?.ToArray() ?? jsonTraversalExpression.Path);
     }
 
+    protected virtual SqlExpression VisitRowValueExpression(
+        PostgresRowValueExpression rowValueExpression,
+        bool allowOptimizedExpansion,
+        out bool nullable)
+    {
+        SqlExpression[]? newValues = null;
+
+        for (var i = 0; i < rowValueExpression.Values.Count; i++)
+        {
+            var value = rowValueExpression.Values[i];
+            var newValue = Visit(value, allowOptimizedExpansion, out _);
```

Comment:
Not sure if you should allow optimized expansion here. @maumar would know better though.